### PR TITLE
Ignore matching data streams if include_data_streams is false

### PR DESCRIPTION
**Pull request by danhermann**
Currently, any requests that match a data stream on APIs that are not enabled for data streams return an error that a data stream was matched. This PR changes the name resolution to simply ignore data streams for APIs that are not enabled for data streams.

Relates to #53100

Relates to https://github.com/elastic/elasticsearch/issues/57712

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/test/indices.data_stream/20_unsupported_apis.yml [hunk 1]
```diff
@@ -61,7 +61,6 @@
         index: logs-foobarbaz
 
   - do:
-      catch: bad_request
```
Comment (martijnvg) [inline rest-api-spec/src/main/resources/rest-api-spec/test/indices.data_stream/20_unsupported_apis.yml:64]: This fails when this yaml test is executed with security enabled.
`logs-*` is expanded to `logs-foobar` and then later this fails with a 404.

So in order to make this test execute constantly with and without security enabled, we should change 
`index: logs-*` to `index: logs-foobar. This should change the meaning of this test.

**Review by martijnvg — APPROVED** [review]
LGTM

**Comment by danhermann** [comment]
This is likely to be abandoned in favor of https://github.com/elastic/elasticsearch/pull/58154. Leaving it open only until that can be confirmed.

**Comment by danhermann** [comment]
@elasticmachine update branch

**Comment by danhermann** [comment]
@elasticmachine run elasticsearch-ci/2

**Pull request merged**